### PR TITLE
Document usage of adding listeners in Traits

### PR DESCRIPTION
**Pull request by basepack**
Credits go to @joshhanley and @booni3 in their comment I found today here: 
https://github.com/livewire/livewire/issues/2381#issuecomment-764022652
https://github.com/livewire/livewire/issues/2381#issuecomment-764805582

### CODE REVIEW

File: events.blade.php [hunk 1]
```diff
@@ -84,6 +84,26 @@ protected function getListeners()
 @endslot
 @endcomponent
 
+### Using listeners in a Trait
+
+You can also add listeners in a Trait. All you need to do is to merge them with the eventual existing listeners of the component in your Trait's mount method:
+
+@component('components.code-component')
+@slot('class')
+class MyTrait {
```
Comment (joshhanley) [inline events.blade.php:93]: Change this to `trait MyTrait`

**Review by joshhanley — CHANGES_REQUESTED** [review]
Looks good! Thanks for PR'ing! Just need to fix the trait definition and should all be good.

Another thought is the `initialize{Trait}` hook should probably also be documented on the traits page, but I think that should be a seperate PR.

File: events.blade.php (hunk 1 above)
Comment (basepack) [reply events.blade.php]: Well there I go with copying and pasting from the docs :)

But it's fixed!

**Comment by calebporzio** [comment]
Thanks for PRing this, but like Josh said, "initializeMyTrait" isn't documented anywhere I don't think. And in fact, I wonder if this would be better to use `mountMyTrait` for less mental overhead? Idk. Thanks!

**Comment by joshhanley** [comment]
@calebporzio it needs to be in `initializeTrait` otherwise you end up with the listeners not being registered on subsequent requests and throws an error, see https://github.com/livewire/livewire/issues/2401. 

I believe initalizeTrait should probably be documented as it ensures that you run code (like registering listeners) on each request regardless of whether it is initial or subsequent requests.

I tested this with @booni3 at the time and couldn't get it to work with mount, but initialize did work.

**Pull request closed without merging**